### DISCUSSION TITLE
인기글에 대한 고찰

### DISCUSSION THREAD

**Discussion by cho-ja-young** [MajorIn-ABO/MajorIn_FE#64] [Ideas]
인기글 관련해서 찾아보다가 읽게 된 글!
https://jongseoung.tistory.com/142

요약하자면, 

> 인기글을 그냥 조회순이나 좋아요수로만 하게 되면 옛날 글이 무조건 핫게로 가버리는 현상이 일어나게 된다
> 따라서 이를 막을 수 있는 방법이 필요

그런데, 진짜 맞는 말이라 이 글쓴이가 주장하는 코드 작성 방법을 우리 프로젝트에 넣으려고 함!
혹시 다른 좋은 아이디어가 있다면 알려주기 바람

**Comment by wlsl10** [comment]
생각지 못한 부분이었는데 블로그에서 통찰력 있게 정리를 잘 해준 것 같아.
인기글에 오래된 글만 뜨거나 인기글이 바뀌지 않고 계속 유지되는 문제가 생길 수 있을 것 같아.
인기글 알고리즘에는 무조건 시간 관련 조건이 있어야할 것 같네!
1번(최근 일주일동안 작성한 게시글만 인기글로 등록이 가능하게 설정)과 3번(인기글에 시간 점수를 매겨 시간이 지날수록 점수가 줄어들도록 설정) 방법이 좋은 것 같은데
학교 커뮤니티는 시간이 지나도 늘 비슷한 주제가 관심을 끄는 것 같아서 3번 Aging 기법을 쓰는 게 좋을 것 같아!
